vsphere/resource_vsphere_virtual_machine.go

```diff
@@ -310,12 +311,15 @@ func resourceVSphereVirtualMachineRead(d *schema.ResourceData, meta interface{})
 	if vprops.ResourcePool != nil {
 		d.Set("resource_pool_id", vprops.ResourcePool.Value)
 	}
-	// Set the folder
-	f, err := folder.RootPathParticleVM.SplitRelativeFolder(vm.InventoryPath)
-	if err != nil {
-		return fmt.Errorf("error parsing virtual machine path %q: %s", vm.InventoryPath, err)
+	// If the VM is part of a vApp, the inventory path will not point to a host
+	// path rather than a VM path, so this step must be skipped.
```

Comment:
I'm having issues understanding this - what does the path of a VM in a vApp container look like then, and how should we be operating on it?

Reply:
I made some changes so the folder isn't skipped when the VM is in a vApp container, but it also cannot be moved to a new folder since it results in an error.

* Is this the inventory path returned by InventoryPath (assuming yes here) - Yes it is.
* Can you see the VM still by govc ls /dc1/vm/ - This does still work
* What happens to the path if you move a VM that belongs to a vApp container to a folder - You cannot. It results in an error.
* Can we make a helper that works off of the above? - Not certain on what you mean here. I did make it so that the virtual machine's VM folder is what `folder` is populated with.